frontend/libs/studio-components-legacy/src/components/StudioTextResourceInput/utils.ts

```diff
@@ -27,6 +26,30 @@ export function changeTextResourceInList(
   );
 }
 
-export function determineDefaultMode(currentId?: string | null): Mode {
-  return currentId ? Mode.EditValue : Mode.Search;
+export function createNewTextResource(value: string): TextResource {
+  return {
+    id: generateRandomTextResourceId(),
+    value,
+  };
+}
+
+export function generateRandomTextResourceId(length: number = 12): string {
+  let result = '';
+  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+
+  for (let i = 0; i < length; i++) {
+    result += characters.charAt(Math.floor(Math.random() * characters.length));
+  }
+
+  return `id_${result}`;
+}
+
+export function getTextResourceValueIfTextResourceExists(
+  textResources: TextResource[],
+  textResource: TextResource,
+  currentId: string,
+): string {
+  const isCurrentIdATextResource: boolean =
+    textResources.filter((item: TextResource) => item.id === currentId).length !== 0;
```

Comment:
Vi kan bruke `some` til å sjekke om et objekt i en liste matcher et gitt predikat:
```suggestion
    textResources.some((item: TextResource) => item.id === currentId);
```
Er `currentId` for øvrig nødvendig som parameter her? Viser til øvrig kommentar om `currentId`.